platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -201,15 +202,15 @@ public BuildOperationDescriptor.Builder description() {
     }
 
     @Override
-    public void registerWatchableHierarchy(File watchableHierarchy) {
+    public void registerWatchableHierarchy(File watchableHierarchy, @Nullable File probeLocation) {
         updateRootUnderLock(currentRoot -> {
             if (watchRegistry == null) {
-                watchableHierarchiesRegisteredEarly.add(watchableHierarchy);
+                watchableHierarchiesRegisteredEarly.put(watchableHierarchy, probeLocation);
```

Comment:
🤔 It seems pretty important to figure out how to avoid passing `null` here. It would make the code much more readable for me.